drivers/state.go

```diff
@@ -116,6 +116,41 @@ func (d *StateDriver) NewAccountActor(addrType address.Protocol, balanceAttoFil
 	return addr, idAddr
 }
 
+func (d *StateDriver) ResolveAddress(addr address.Address) address.Address {
+	if addr.Protocol() == address.ID {
+		return addr
+	}
+	for k, v := range d.actorIDMap {
+		if v == addr {
+			return k
+		}
+	}
+	return address.Undef
+}
+
+func (d *StateDriver) AddrEquals(addr1, addr2 string) bool {
```

Comment:
1. This is more fallout from changing the MessageReceipt structure to be non-canonical, but
2. Why don't you just compare strings?